llvm/test/Transforms/LoopVectorize/AArch64/sve-structured-store.ll

```diff
@@ -0,0 +1,279 @@
+; RUN: opt -mtriple=aarch64 -mattr=+sve -S -passes=loop-vectorize,interleaved-access  -enable-masked-interleaved-mem-accesses=true -sve-gather-overhead=10 -sve-scatter-overhead=10  -enable-shuffle-padding=true -force-vector-width=16 -aarch64-sve-vector-bits-min=512 < %s -o - | FileCheck %s --check-prefixes=ENABLE-SHUFFLE-PADDING
+; RUN: opt -mtriple=aarch64 -mattr=+sve -S -passes=loop-vectorize,interleaved-access  -enable-masked-interleaved-mem-accesses=true -sve-gather-overhead=10 -sve-scatter-overhead=10  -enable-shuffle-padding=false -force-vector-width=16  -aarch64-sve-vector-bits-min=512 < %s -o - | FileCheck %s --check-prefixes=DISABLE-SHUFFLE-PADDING
+
+%struct.patic = type { float, float, float }
+
+; for (int i = 0; i < num; i++) {
+;   ps[i].x = factor * ps[i].x;
+;   ps[i].y = factor * ps[i].y;
+; }
+;
+; Function Attrs: argmemonly mustprogress nofree norecurse nosync nounwind uwtable vscale_range(2,2)
+define void @test(i32 noundef %num, ptr nocapture noundef %ps) {
+; ENABLE-SHUFFLE-PADDING-LABEL: @test(
```

Comment:
The ENABLE and DISABLE check lines are identical so I don't know what you're trying to show here that's different? What effect does your patch have on these tests? At the moment it's hard to know the impact of your code change.

Reply:
Hi, @david-arm, check lines for ENABLE and DISABLE are different. When the shuffle padding is enabled, the interleaved store will be transformed into `llvm.aarch64.sve.st3`.  And if the feature is disabled, there is no `sve.st3`, and I use `ISABLE-SHUFFLE-PADDING-NOT` here.

Reply:
My apologies, I see that now! In that case, I think it's worth having more testing than just the existence of st3 in the IR for the ENABLE-SHUFFLE case. I believe it's important to show the input to the store as well, i.e. how the stored value is generated.

Reply:
Done, but it seems that the file has become very large...